coretrace/simulation_data/element.hpp

```diff
@@ -527,7 +537,7 @@ class ElementBase : public Element
   Matrix3d local_to_reference;
 
   // element_ptr reference_element;
-  Element *reference_element;
+  Element *reference_element; // todo: this is a raw pointer, probably shouldn't be
```

Comment:
This is actually intentional. The reference element should always exist and getting access to the shared pointer hierarchy is difficult when this is set.